packages/core/src/reporters/json/json-reporter.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from 'path';
+
+import fileUrl = require('file-url');
+
+import { Logger } from '@stryker-mutator/api/logging';
+import { mutationTestReportSchema, Reporter } from '@stryker-mutator/api/report';
+import { StrykerOptions } from '@stryker-mutator/api/core';
+import { commonTokens, tokens } from '@stryker-mutator/api/plugin';
+
+import { mkdir, deleteDir, writeFile } from './json-reporter-util';
+
+const DEFAULT_BASE_FOLDER = path.normalize('reports/mutation/json');
+
+export default class JsonReporter implements Reporter {
+  private _baseDir!: string;
+  private mainPromise: Promise<void> | undefined;
+
+  constructor(private readonly options: StrykerOptions, private readonly log: Logger) {}
+
+  public static readonly inject = tokens(commonTokens.options, commonTokens.logger);
+
+  public onMutationTestReportReady(report: mutationTestReportSchema.MutationTestResult) {
+    this.mainPromise = this.generateReport(report);
+  }
+
+  public wrapUp() {
+    return this.mainPromise;
+  }
+
+  public async generateReport(report: mutationTestReportSchema.MutationTestResult) {
+    const indexFileName = path.resolve(this.baseDir, 'report.json');
+
+    await this.cleanBaseFolder();
+    await writeFile(indexFileName, JSON.stringify(report));
+
+    this.log.info(`Your report can be found at: ${fileUrl(indexFileName)}`);
+  }
+
+  private get baseDir(): string {
```

Comment:
As stated earlier, this null check is no longer needed if you use `"default": {}` for the `jsonReporter` option. The base dir inside the `StrykerOptions` will always be filled with a string.